packages/nvidia_nat_eval/tests/eval/dataset_handler/test_dataset_downloader.py

```
@@ -0,0 +1,67 @@
+        downloader.download_with_signed_url("https://example.com/dataset.json", str(tmp_path / "dataset.json"))
+
+
+def test_s3_download_missing_boto3_has_install_hint(monkeypatch: pytest.MonkeyPatch, tmp_path: Path) -> None:
+    monkeypatch.setitem(sys.modules, "boto3", None)
+    monkeypatch.setitem(sys.modules, "botocore", None)
+    monkeypatch.setattr(sys, "meta_path", [_BlockModules({"boto3", "botocore"}), *sys.meta_path])
+
+    config = EvalDatasetJsonConfig(
+        file_path=tmp_path / "dataset.json",
+        remote_file_path="dataset.json",
+        s3=EvalS3Config(bucket="bucket", access_key="access-key", secret_key="secret-key"),
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "secret_key"

(S106)